### DISCUSSION TITLE
Which one is better call params/queries from loader or client side ?

### DISCUSSION THREAD

**Discussion by robyhuzwandar** [remix-run/remix#4387] [Q&A] [answered]
I just thinking to manage all data from the server side so that the client only receives data, including params data and queries from routes.

but I don't find any compelling reason to do it other than just cleaner code, and if the server and client need the same params then just call params once and send it to the client.

**Comment by kiliman** [comment, marked as answer]
If you're referring to the route params, then check out the `useParams` hook to use in client code. https://remix.run/docs/en/v1/api/conventions#dynamic-route-parameters

As for why data fetching in loaders... besides eliminating network waterfalls, you also can reduce client bundle since any fetching libraries (like GraphQL, etc) don't need to be sent to the browser.

Also it adds to consistency. I would only revert to client side fetching if your use case calls for it.
